server/attr/attributeTracker.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package attr
+
+import (
+	mixerpb "istio.io/mixer/api/v1"
+)
+
+// AttributeTracker is responsible for tracking a set of live attributes over time.
+//
+// An instance of this type is created for every gRPC stream incoming to the
+// mixer. The instance tracks a current dictionary along with a set of
+// attribute contexts.
+type AttributeTracker interface {
```

Comment:
Are we expecting multiple impls of attribute tracker?
If not this interface is not needed.